src/core/execute.c

```diff
@@ -2461,6 +2511,39 @@ static int setup_exec_directory(
                         goto fail;
         }
 
+        /* First, ensure we are not asked to created symlinks for exec dirs that do not exist. */
+        r = prune_non_existing_exec_directories_symlinks(context, params, type);
+        if (r < 0) {
+                log_error_errno(r, "Failed to normalize special execution directory symlinks in %s: %m", params->prefix[type]);
+                goto fail;
+        }
+
+        /* If we are not going to run in a namespace, set up the symlinks - otherwise
+         * they are set up later, to allow configuring empty var/run/etc. */
+        if (!needs_mount_namespace) {
+                char **src, **dst;
+
+                STRV_FOREACH_PAIR(src, dst, context->directories[type].symlinks) {
+                        _cleanup_free_ char *src_abs = NULL, *dst_abs = NULL;
+
+                        src_abs = path_join(params->prefix[type], *src);
+                        dst_abs = path_join(params->prefix[type], *dst);
+                        if (!src_abs || !dst_abs) {
+                                r = -ENOMEM;
+                                goto fail;
+                        }
+
+                        r = mkdir_parents_label(dst_abs, 0755);
+                        if (r < 0)
+                                goto fail;
+
+                        r = symlink_idempotent(src_abs, dst_abs, true);
+                        if (r < 0)
+                                goto fail;
+                }
```

Comment:
unless i am mistaken the earlier commit adds preitty much the same loop for the "private" symlinks. implement a common helper func for the two cases?